core/operation.ts

```diff
@@ -1,34 +1,45 @@
-import {
-  IColumnsDescriptor,
-  mapToColumnProtoArray,
-  Resolvable,
-  Session
-} from "@dataform/core/session";
+import { mapToColumnProtoArray, Session } from "@dataform/core/session";
 import * as utils from "@dataform/core/utils";
 import { dataform } from "@dataform/protos";
-
-export type OContextable<T> = T | ((ctx: OperationContext) => T);
-
-export interface OConfig {
-  dependencies?: Resolvable | Resolvable[];
-  tags?: string[];
-  description?: string;
-  columns?: IColumnsDescriptor;
+import {
+  IColumnsDescriptor,
+  ICommonContext,
+  ICommonOutputConfig,
+  Resolvable
+} from "@dataform/core/common";
+import { Contextable } from "@dataform/core/contextable";
+
+export interface IOperationConfig extends ICommonOutputConfig {
+  /**
+   * Tells Dataform whether this operations action creates an output table
+   * and is allowed to be referenced using the <code>ref</code> function.
+   *
+   * When set to true, the SQL operations should create a table or view of the given name.
+   * This can be accomplished using the <code>self()</code> context function.
+   *
+   * For example, an operation with output typically contains SQL such as:
+   * ```sql
+   * create or replace table ${self()} as select ...
+   * ```
+   */
   hasOutput?: boolean;
-  database?: string;
-  schema?: string;
 }
 
+export interface IOperationContext extends ICommonContext {}
```

Comment:
ditto with the confusing `dataset` terminology caused by extending `ICommonContext`, can we override some of the docs, or perhaps make the docs in `ICommonContext` a bit more generic?

Reply:
Made the common descriptions a bit more generic!